pkg/config/config_parser.go

```diff
@@ -141,7 +144,7 @@ func decodeConfig(r io.Reader) (*Config, diag.Diagnostics) {
 		if len(errs) == 0 {
 			return nil, diag.FromError(errors.New("Failed to validate config with schema"), diag.USER, diag.WithSummary("Invalid configuration"))
 		}
-		diags := diag.Diagnostics{}
+		var diags diag.Diagnostics
```

Comment:
good change, I think this would have returned `error: no errors`

Reply:
nil slice is valid (and empty) slice so both would give us `error: no error` if cast to `error`.